feat: index public call requests from pending transactions

Mautjee opened the pull request:
Add support for extracting and storing public call requests from pending transactions with new API endpoints

Mautjee committed:
storing public contract call

Mautjee committed:
refactor to seperate controllers

Mautjee committed:
update route for api

Mautjee committed:
delete data of dropped tx

FilipHarald commented on services/explorer-api/migrations/0000_nifty_moonstone.sql:
```diff
@@ -14,9 +14,16 @@ CREATE TABLE IF NOT EXISTS "aztec-chain-connection" (
 	"protocol_contract_addresses" jsonb NOT NULL
```
I can see that you have deleted all the older migrations. We can't do that any longer since we need to persist the DB

FilipHarald commented on services/explorer-api/src/svcs/http-server/routes/paths_and_validation.ts:
```diff
@@ -42,6 +42,10 @@ export const paths = {
   txs: "/l2/txs",
   txByHash: `/l2/txs/:${txEffectHash}`,
 
+  publicDataTx: `/l2/public-call-requests/tx/:${txEffectHash}`,
```
Since public-call-requests is only one DB-table. We should also only have one (or two) endpoints.

```ts
publicCallRequestsByTxHash: /l2/public-call-requests/:${txEffectHash}
publicCallRequests: /l2/public-call-requests // this one can have option-params to search for contractAddress or senderAddress
```

FilipHarald commented on services/explorer-api/src/svcs/database/schema/l2public-call/index.ts:
```diff
@@ -0,0 +1,24 @@
+import { l2Tx } from "../l2tx/index.js";
+
+export const l2TxPublicCallRequest = pgTable("tx_public_call_request", {
+  id: varchar("id").notNull().primaryKey(),
```
Is there a reason you are using id and not txHash here? I think txHash would be better.

Mautjee replied:
Oke, I will change this

Mautjee committed:
Merge branch 'main' into feature/index-public-data-from-pending-tx

Mautjee committed:
added migration

Mautjee committed:
fix bug for api route

Mautjee committed:
merge: bring feature branch up to date with main (v4)

Mautjee committed:
feat: index public tx data from pending transactions

- Add callType (non_revertible/revertible/teardown) to PublicCallRequest schema
- Extend ChicmozL2PendingTx with gas settings, gas used, expiration timestamp,
  fee payment method, and summary counts (note hashes, nullifiers, L2->L1 msgs, private logs)
- Extract teardown call request alongside non-revertible and revertible calls
- Fix await bug: tx.getTxHash() is synchronous in v4
- Add new columns to l2Tx DB table for all new fields
- Add call_type enum column to tx_public_call_request table
- Fix typo: getPublicCallRequessByAddressSchema -> getPublicCallRequestsByAddressSchema
- Update OpenAPI schemas to include callType field
- Generate migration 0011_flippant_thunderbolt for new table and columns

Mautjee committed:
opencode package lock update

Mautjee committed:
Merge branch 'main' into feature/index-public-data-from-pending-tx

Mautjee committed:
fix: correct authwit and enable L1/L2 messaging in event-cannon

- Replace SetPublicAuthwitContractInteraction with wallet.createAuthWit using
  CallIntent (call + getFunctionCall) for burn_private — private functions
  require a local auth witness, not an on-chain public authwit
- Use .with({ authWitnesses }) instead of .withAuthWit (correct API)
- Enable SCENARIO_L1L2_PUBLIC_MESSAGING and SCENARIO_L1L2_PRIVATE_MESSAGING
- Set INIFINITE_LOOP=true to prevent CrashLoopBackOff in k8s

Mautjee committed:
small changes for running sandbox

Mautjee committed:
add extra data for storing l2 to l1 message data

Mautjee committed:
new scenario for FCP contract

Mautjee committed:
Additions of public call data to existing pages

Mautjee committed:
address pages

Mautjee committed:
adding table view to detail pages pills

Mautjee committed:
introduces the data table to more places where public call requests are shown

Mautjee committed:
update l1 address activity

Mautjee committed:
pending tx now link to addresses

Mautjee committed:
update rollup version select

Mautjee committed:
show fee payer and initiator in mined tx

Mautjee committed:
artifact matching with public contract calls

Mautjee committed:
Merge branch 'feature/index-public-data-from-pending-tx' into feature(frontend)/index-public-data-from-pending-tx

Mautjee committed:
update migration file

Mautjee committed:
Merge branch 'feature/index-public-data-from-pending-tx' into feature(frontend)/index-public-data-from-pending-tx

Mautjee committed:
update the tooltip

Mautjee committed:
add the correct way of handling function name selector

Mautjee committed:
Merge branch 'main' into feature/index-public-data-from-pending-tx

Mautjee commented on services/explorer-api/src/svcs/database/controllers/l2Public-call/store.ts:
```diff
@@ -0,0 +1,88 @@
+const resolveArtifactNames = async (
+  contractAddress: string,
+  functionSelector: string | undefined,
+): Promise<{ contractName: string | null; functionName: string | null }> => {
+  const rows = await db()
+    .select({
+      artifactContractName: l2ContractClassRegistered.artifactContractName,
+      selectorMap: l2ContractClassRegistered.selectorMap,
+    })
+    .from(l2ContractInstanceDeployed)
+    .leftJoin(
+      l2ContractClassRegistered,
+      eq(
+        l2ContractInstanceDeployed.currentContractClassId,
+        l2ContractClassRegistered.contractClassId,
+      ),
+    )
```
@copilot apply changes based on [this feedback](https://github.com/aztec-scan/chicmoz/pull/540#discussion_r3136704705)

Copilot committed:
fix: consolidate public call endpoints and fix resolveArtifactNames join on version

Copilot committed:
fix: return 400 when both contractAddress and senderAddress are provided

FilipHarald commented on services/aztec-listener/src/events/emitted/on-pending-txs.ts:
```diff
@@ -1,20 +1,198 @@
+  }
+
+  // Build a lookup map calldataHash → functionSelector from the calldata-enriched requests.
+  // TODO: ABI decoding — map functionSelector to human-readable function name + decoded params.
```
Should TODO be here?

Mautjee committed:
testnet env

FilipHarald commented on services/explorer-api/src/svcs/database/controllers/l2contract/source-verification.ts:
```diff
@@ -84,10 +84,11 @@ export const updateSourceVerificationJobStatus = async ({
+    status: status as (typeof sourceVerificationJobs.$inferInsert)["status"],
     error: error ?? null,
-    failureStage: failureStage ?? null,
+    failureStage: (failureStage ??
```
Erm, don't know why this change is here?

FilipHarald replied:
If it is the same as main then there should be no changes to this file - right?

Mautjee replied:
@copilot why is there a diff when it should be the same as main?

Mautjee committed:
remove todo

Copilot committed:
fix: address review feedback - SQL filter, batch query, loading state, Link, import type

Co-authored-by: Mautjee <19729404+Mautjee@users.noreply.github.com>

Copilot committed:
fix: restore source-verification.ts to match main (remove merge artifact)

Agent-Logs-Url: https://github.com/aztec-scan/chicmoz/sessions/0ff603d5-271e-4369-95d0-6f899c1b13d8

Co-authored-by: Mautjee <19729404+Mautjee@users.noreply.github.com>
